### DISCUSSION TITLE
Balance Power not always equal to sum of monitored circuits

### DISCUSSION THREAD

**Discussion by matt1131** [emporia-vue-local/esphome#186] [Q&A] [answered]
I noticed after installing this that the Balance Power doesn't always equal the sum of the circuits that are monitored.

The important colors below are:
Light Blue = Total Power
Orange = Balance Power

![image](https://github.com/emporia-vue-local/esphome/assets/24827404/dcf26bee-7c7f-437a-9c88-86980ddc03ee)

Notice how the darker blue and pink changes aren't picked up by Balance Power. Or how Balance Power drops entirely below the green circuit.

My configuration is: 

    name: "Balance Power"
    lambda: !lambda |-
      return max(0.0f, id(total_power).state -
        id( cir1).state -
        id( cir2).state -
        id( cir3).state -
        id( cir4).state -
        id( cir5).state -
        id( cir6).state -
        id( cir7).state -
        id( cir8).state -
        id( cir9).state -
        id(cir10).state -
        id(cir11).state -
        id(cir12).state -
        id(cir13).state -
        id(cir14).state -
        id(cir15).state -
        id(cir16).state);
    update_interval: 2.88s
    id: balance_power
    device_class: power
    state_class: measurement
    unit_of_measurement: "W"

What should I change in my configuration to get a more accurate reading?

**Comment by flaviut** [comment]
The Balance Power field is simple arithmetic--if you're not using all 16 current clamps, you'll need to remove those clamps from the math expression.

If incorrect data is leading to this situation instead, you'll need to look at your system & see which fields seem unreasonable & perhaps use a current clamp multimeter to calibrate.

**Comment by matt1131** [comment, marked as answer]
I realized I'm an idiot. Balance power is not a sum of all monitored circuits, it's total power minus the sum of all monitored circuits (i.e. everything else).
